### PR TITLE
feat: update support surface building

### PR DESCRIPTION
**Pull request by asalzburger**
This PR: 

- changes how the definition of `ProtoSupport` is done
- renames the `SupportHelper` to `SupportSurfacesHelper`
- adds rectangular support shapes to `SupportSurfacesHelper` and `LayerStructureBuilder`
- extends the Unit tests

### CODE REVIEW

**Comment by asalzburger** [comment]
Hi @Corentin-Allaire - I requested this one for you. This is the way how to automatically build support (i.e. passive) material surfaces in the new geometry model, you can tell via a `ProtoSupport` class where to put it, or provide it directly.

I will have a follow-up PR which shows this on the Open Data Detector.

**Review by Corentin-Allaire — COMMENTED** [review]
Look good to me ! I have noted some small mistake in the comments but except from that it looks good

File: Core/include/Acts/Detector/detail/SupportSurfacesHelper.hpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  ///
+  /// @param lExtent the layer and/or volume extent
+  ///
+  /// @return the support surface components
+  SupportSurfaceComponents operator()(const Extent& lExtent) const;
+};
+
+/// @brief Helper method to build planar support structure
+struct RectangularSupport {
+  /// Placement - the remaining loc0, loc1 are then cyclcic
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Core/include/Acts/Detector/detail/SupportSurfacesHelper.hpp]: It's indeed cyclic - you chose a placement direction, the remaining then are clear by cyclic order

**Review by Corentin-Allaire — APPROVED** [review]

**Comment by asalzburger** [comment]
`linux-nodep` has still the timeout to the server ... overriding the blocking CI here.

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [b0432e5325706e0ff90f14521294c9528505c9ae]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522034)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsEFTrackFit.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522033)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522032)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522031)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522030)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522029)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522028)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522027)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522026)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522025)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522024)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522023)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522022)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522021)||
|:green_circle: | [run_ci_tests: ../athena/Tracking/Acts/ActsConfig/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522020)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522019)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522018)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522017)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522016)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522015)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522014)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522013)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522012)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35522011)||

**Pull request merged**